### PR TITLE
Question

### PR DESCRIPTION
**Pull request by helinsv**

### CODE REVIEW

File: src/app/rest/rest.component.html [hunk 1]
```diff
@@ -0,0 +1,3 @@
+<p>
```
Comment (nsenkevich) [inline src/app/rest/rest.component.html]: why we have this file?

File: src/app/rest/rest.component.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import { Component, OnInit } from '@angular/core';
```
Comment (nsenkevich) [inline src/app/rest/rest.component.ts]: why we have this file?

File: src/app/rest/rest.component.spec.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (nsenkevich) [inline src/app/rest/rest.component.spec.ts]: why we have this file?

File: src/app/app.module.ts [hunk 4]
```diff
@@ -1,18 +1,26 @@
...
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { RestComponent } from './rest/rest.component';
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    RestComponent
```
Comment (nsenkevich) [inline src/app/app.module.ts]: where you using this component RestComponent?

File: src/app/app.component.ts [hunk 5]
```diff
@@ -7,4 +11,25 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'poll';
+  private apiUrl = 'http://localhost:3000/api/questions';
+  data: any = {};
+
+  constructor(private http: Http) {
+    console.log('hello');
```
Comment (nsenkevich) [inline src/app/app.component.ts]: remove console.log

File: src/app/app.component.ts [hunk 6]
```diff
@@ -7,4 +11,25 @@ import { Component } from '@angular/core';
...
+
+  getData(){
+    return this.http.get(this.apiUrl)
+    .pipe(map(r => r.json()))
+      
+  }
+
+  getQuestions(){
+    this.getData().subscribe(data => {
+      console.log(data);
```
Comment (nsenkevich) [inline src/app/app.component.ts]: remove console.log

File: src/app/app.component.ts [hunk 7]
```diff
@@ -7,4 +11,25 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'poll';
+  private apiUrl = 'http://localhost:3000/api/questions';
+  data: any = {};
```
Comment (nsenkevich) [inline src/app/app.component.ts]: you should never use any, please provide type

File: src/app/app.component.ts [hunk 8]
```diff
@@ -7,4 +11,25 @@ import { Component } from '@angular/core';
 })
 export class AppComponent {
   title = 'poll';
+  private apiUrl = 'http://localhost:3000/api/questions';
```
Comment (nsenkevich) [inline src/app/app.component.ts]: move to environment

File: src/app/app.component.ts [hunk 9]
```diff
@@ -7,4 +11,25 @@ import { Component } from '@angular/core';
...
+  }
+
+  getData(){
+    return this.http.get(this.apiUrl)
+    .pipe(map(r => r.json()))
+      
+  }
+
+  getQuestions(){
+    this.getData().subscribe(data => {
```
Comment (nsenkevich) [inline src/app/app.component.ts]: you can use |async

File: src/app/app.component.html [hunk 10]
```diff
@@ -1,21 +1,22 @@
...
-  </li>
-</ul>
 
+<div *ngFor="let question of data">
+  <h3>{{question.id}}</h3>
+  <h2>{{question.question}}</h2>  
+</div>
+
+<h1>Blog Posts</h1>
+<div *ngIf="undefined === restItems">Loading...</div>
```
Comment (nsenkevich) [inline src/app/app.component.html]: you can use |async

File: src/app/app.component.html [hunk 11]
```diff
@@ -1,21 +1,22 @@
...
+<div *ngIf="undefined !== restItems">
```
Comment (nsenkevich) [inline src/app/app.component.html]: restItems not good name use answers

Comment (nsenkevich) [reply src/app/app.component.html]: restItems[''] why?

File: src/app/app.component.html [hunk 12]
```diff
@@ -1,21 +1,22 @@
...
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://angular.io/cli">CLI Documentation</a></h2>
-  </li>
-  <li>
-    <h2><a target="_blank" rel="noopener" href="https://blog.angular.io/">Angular blog</a></h2>
-  </li>
-</ul>
 
+<div *ngFor="let question of data">
```
Comment (nsenkevich) [inline src/app/app.component.html]: data should be questions

File: package.json [hunk 13]
```diff
@@ -16,9 +16,11 @@
     "@angular/compiler": "~7.1.0-beta.0",
     "@angular/core": "~7.1.0-beta.0",
     "@angular/forms": "~7.1.0-beta.0",
+    "@angular/http": "^7.0.3",
     "@angular/platform-browser": "~7.1.0-beta.0",
     "@angular/platform-browser-dynamic": "~7.1.0-beta.0",
     "@angular/router": "~7.1.0-beta.0",
+    "@ng-bootstrap/ng-bootstrap": "^4.0.0",
```
Comment (nsenkevich) [inline package.json]: you added bootstrap why you did't used

File: src/app/app.component.ts [hunk 14]
```diff
@@ -32,8 +32,31 @@ export class AppComponent {
       this.data = data
     })
   }
+
+  votes: number = 0; 
+  procent: number = 0; 
+  message: string = 'Виберіть кандадата';
+ 
+  clickVote(a){
```
Comment (nsenkevich) [inline src/app/app.component.ts]: a? what a means

File: src/app/app.component.ts [hunk 15]
```diff
@@ -32,8 +32,31 @@ export class AppComponent {
...
+    var votes = this.votes++;
+    var vote =  a.vote++ 
+    this.procent = vote*100/votes;
```
Comment (nsenkevich) [inline src/app/app.component.ts]: procent  maybe percentage ?

File: src/app/app.component.ts [hunk 16]
```diff
@@ -32,8 +32,31 @@ export class AppComponent {
...
+  procent: number = 0; 
+  message: string = 'Виберіть кандадата';
+ 
+  clickVote(a){
+    var votes = this.votes++;
+    var vote =  a.vote++ 
+    this.procent = vote*100/votes;     
+  }
+
+  getValue(array, votes) {
```
Comment (nsenkevich) [inline src/app/app.component.ts]: array newer call property with type name use votes

File: src/app/app.component.ts [hunk 17]
```diff
@@ -1,4 +1,8 @@
 import { Component } from '@angular/core';
```
Comment (nsenkevich) [inline src/app/app.component.ts]: move logic from app component to question component and answer component

File: src/app/app.component.ts [hunk 18]
```diff
@@ -0,0 +1,62 @@
...
+import { map } from 'rxjs/operators';
+
+
+@Component({
+  selector: 'app-root',
+  templateUrl: './app.component.html',
+  styleUrls: ['./app.component.sass']
+})
+export class AppComponent {
+  title = 'poll';
```
Comment (nsenkevich) [inline src/app/app.component.ts]: why?

File: src/app/app.component.ts [hunk 19]
```diff
@@ -0,0 +1,62 @@
...
+    this.procent = vote*100/votes;     
+  }
+
+  getValue(array, votes) {
+    for (var i = 0; i < array.length; i++) {
+      array[i].procent = array[i].vote * 100 / votes;
+    } 
+  } 
+}
+
```
Comment (nsenkevich) [inline src/app/app.component.ts]: why we have all this empty lines?

File: .vs/VSWorkspaceState.json [hunk 20]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (nsenkevich) [inline .vs/VSWorkspaceState.json]: .vs files to ignore please?

File: src/app/app.component.ts (hunk 15 above)
Comment (nsenkevich) [reply src/app/app.component.ts]: var you should use let or const

File: src/app/rest/rest.component.html (hunk 1 above)
Comment (helinsv) [reply src/app/rest/rest.component.html]: deleted

File: src/app/rest/rest.component.ts (hunk 2 above)
Comment (helinsv) [reply src/app/rest/rest.component.ts]: deleted

File: src/app/rest/rest.component.spec.ts (hunk 3 above)
Comment (helinsv) [reply src/app/rest/rest.component.spec.ts]: deleted

File: src/app/app.module.ts (hunk 4 above)
Comment (helinsv) [reply src/app/app.module.ts]: deleted

File: src/app/app.component.ts (hunk 5 above)
Comment (helinsv) [reply src/app/app.component.ts]: done

File: src/app/app.component.ts (hunk 6 above)
Comment (helinsv) [reply src/app/app.component.ts]: done

File: src/app/app.component.html (hunk 11 above)
Comment (helinsv) [reply src/app/app.component.html]: its old code

File: src/app/app.component.html (hunk 10 above)
Comment (helinsv) [reply src/app/app.component.html]: its old code

File: src/app/app.component.ts (hunk 7 above)
Comment (helinsv) [reply src/app/app.component.ts]: string

File: src/app/app.component.ts (hunk 14 above)
Comment (helinsv) [reply src/app/app.component.ts]: changed

File: src/app/app.component.ts (hunk 15 above)
Comment (helinsv) [reply src/app/app.component.ts]: changed

File: .vs/VSWorkspaceState.json (hunk 20 above)
Comment (helinsv) [reply .vs/VSWorkspaceState.json]: # Dependency directories
node_modules/
jspm_packages/
.vs/ 

ok?

File: src/app/app.component.ts (hunk 19 above)
Comment (helinsv) [reply src/app/app.component.ts]: must be }}}? Dont understand

File: package.json (hunk 13 above)
Comment (helinsv) [reply package.json]: Dont delete this package.

core.js:14576 ERROR Error: The module AppModule was bootstrapped, but it does not declare "@NgModule.bootstrap" components nor a "ngDoBootstrap" method. Please define one of these.
    at PlatformRef.push../node_modules/@angular/core/fesm5/core.js.PlatformRef._moduleDoBootstrap (core.js:16650)
    at core.js:16612
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:16135)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:387)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:138)
    at zone.js:872
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:16126)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
defaultErrorLogger @ core.js:14576
main.ts:12 Error: The module AppModule was bootstrapped, but it does not declare "@NgModule.bootstrap" components nor a "ngDoBootstrap" method. Please define one of these.
    at PlatformRef.push../node_modules/@angular/core/fesm5/core.js.PlatformRef._moduleDoBootstrap (core.js:16650)
    at core.js:16612
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:388)
    at Object.onInvoke (core.js:16135)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invoke (zone.js:387)
    at Zone.push../node_modules/zone.js/dist/zone.js.Zone.run (zone.js:138)
    at zone.js:872
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:421)
    at Object.onInvokeTask (core.js:16126)
    at ZoneDelegate.push../node_modules/zone.js/dist/zone.js.ZoneDelegate.invokeTask (zone.js:420)
(anonymous) @ main.ts:12

File: src/app/app.component.ts (hunk 18 above)
Comment (helinsv) [reply src/app/app.component.ts]: Dont understand question.

File: src/app/app.component.ts (hunk 17 above)
Comment (helinsv) [reply src/app/app.component.ts]: done

File: src/app/app.component.ts (hunk 19 above)
Comment (helinsv) [reply src/app/app.component.ts]: done

File: src/app/app.component.ts (hunk 9 above)
Comment (helinsv) [reply src/app/app.component.ts]: add "subscribe(async data..."

**Pull request merged**